moler/device/textualdevice.py

```diff
@@ -90,20 +93,76 @@ def __init__(self, sm_params=None, name=None, io_connection=None, io_type=None,
         # TODO: Need test to ensure above sentence for all connection
         self.io_connection.notify(callback=self.on_connection_made, when="connection_made")
         self.io_connection.notify(callback=self.on_connection_lost, when="connection_lost")
-        self.io_connection.open()
 
         self._cmdnames_available_in_state = dict()
         self._eventnames_available_in_state = dict()
+        self._default_prompt = re.compile(r'^[^<]*[\$|%|#|>|~]\s*$')
+        self._neighbour_devices = None
+        self._open = False
+        msg = "Created device '{}' as instance of class '{}.{}'.".format(
+            self.name,
+            self.__class__.__module__,
+            self.__class__.__name__,
+        )
+        self._log(level=logging.DEBUG, msg=msg)
+
+    def establish_connection(self):
+        """
+        Establishes real connection to device. You have to call this method before device is full operable.
+
+        :return: None
+        """
+        if self._open:
+            return
+        self.io_connection.open()
 
         self._collect_cmds_for_state_machine()
         self._collect_events_for_state_machine()
         self._run_prompts_observers()
-        self._default_prompt = re.compile(r'^[^<]*[\$|%|#|>|~]\s*$')
-        msg = "Created device '{}' as instance of class '{}.{}' with prompts:".format(self.name,
-                                                                                      self.__class__.__module__,
-                                                                                      self.__class__.__name__)
+
+        msg = "Established connection to device '{}' (as instance of class '{}.{}') with prompts: '{}'.".format(
+            self.name,
+            self.__class__.__module__,
+            self.__class__.__name__,
+            self._state_prompts
+        )
+        self._open = True
         self._log(level=logging.INFO, msg=msg)
-        self._log(level=logging.INFO, msg=self._state_prompts)
+
+    def is_open(self):
```

Comment:
Maybe `def is_established` is better name? In this place we check that connection to device is established,